website_sale_product_minimal_price/models/product_template.py

```diff
@@ -1,120 +1,96 @@
-                )['price']
-            )[:1]
-
-            res = cheaper_variant._get_combination_info_variant(pricelist=pricelist)
-
-            combination_info.update({
-                'price': res.get('price'),
-                'list_price': res.get('list_price'),
-                'has_discounted_price': res.get('has_discounted_price'),
-                'has_distinct_price': True,
-            })
-        return combination_info
+        # Inject the boolean for telling if putting "From " text before price
+        res["has_distinct_price"] = has_distinct_price
+        return res
 
     def _get_first_possible_combination(
             self, parent_combination=None, necessary_values=None):
-        """
-        Get the cheaper product combination for the product for website view.
-        We only take into account attributes that generate variants and
-        products with more than one variant.
-        """
-        combination = super()._get_first_possible_combination(
+        """Get the cheaper product combination for the website view."""
+        res = super()._get_first_possible_combination(
             parent_combination=parent_combination,
             necessary_values=necessary_values
         )
-        if (self.env.context.get('website_id') and
+        context = self.env.context
+        if (context.get("website_id") and context.get("pricelist") and
                 self.product_variant_count > 1):
-            ptav_obj = self.env['product.template.attribute.value']
-            pav = self.product_variant_ids.sorted(
-                'website_price')[:1].attribute_value_ids
-            cheaper_combination = ptav_obj.search([
-                ('product_tmpl_id', '=', self.id),
-                ('product_attribute_value_id', 'in', pav.ids),
-            ])
-            variant_combination = combination.filtered(
-                lambda x: x.attribute_id.create_variant == 'always')
-            combination_returned = cheaper_combination + (
-                combination - variant_combination)
-            # Keep order to avoid This combination does not exist message
-            return combination_returned.sorted(
-                lambda x: x.attribute_id.sequence)
-        return combination
+            # It only makes sense to change the default one when there are
+            # more than one variants and we know the pricelist
+            pricelist = self.env["product.pricelist"].browse(
+                context["pricelist"])
+            product_id = self._get_cheapest_info(pricelist)[0]
+            product = self.env["product.product"].browse(product_id)
+            ptavs = product.product_template_attribute_value_ids
+            variant_attributes = ptavs.mapped("attribute_id")
+            # remove returned values that are variant specific
+            res.filtered(lambda x: x.attribute_id not in variant_attributes)
+            # and inject cheapest variant ones
+            res += ptavs
```

Comment:
This is not being sorted. See https://github.com/OCA/e-commerce/pull/460/files#diff-4df29720f5325f6a08d6648efb0e935de9b9cf38bc79e27087fb4c3e8f958c69L118

Why previous code was not correct (filtering if you want if pricelist is present).